src/lib/externaldata.py

```diff
@@ -137,36 +137,41 @@ def update(self):
 class ExternalDataFinishArg(ExternalData):
     PREFIX = '--extra-data='
 
-    def __init__(self, source_path, finish_args, index):
-        arg = finish_args[index]
-
+    def __init__(self, source_path, arg, finish_args):
         # discard '--extra-data=' prefix from the string
         extra_data = arg[len(self.PREFIX) + 1:]
         name, sha256sum, size, _install_size, url = extra_data.split(":", 4)
         data_type = ExternalData.Type.EXTRA_DATA
 
         super().__init__(data_type, source_path, finish_args, name, url, sha256sum, size, [])
 
-        self.index = index
+        self.source = arg[:]
 
     @classmethod
     def from_args(cls, source_path, finish_args):
         return [
-            cls(source_path, finish_args, i)
-            for i, arg in enumerate(finish_args)
+            cls(source_path, arg, finish_args)
+            for arg in finish_args
             if arg.startswith(cls.PREFIX)
         ]
 
     def update(self):
         if self.new_version is not None:
+            if not self.source in self.source_parent:
+                log.warning("Unable to find index for %s", self.source)
```

Comment:
This should only happen on programmer error – I'd just leave this check out and allow `index()` to throw an exception.

Reply:
Ack.

Reply:
Done.